### PR TITLE
Define exported fields and update to percentage fields

### PR DESCRIPTION
**Pull request by monicasarbu**
Create etc/fields.yml to define the exported fields. The yml file is used to generate the documentation and the Elasticsearch template. 

In Kibana, to display a percentage value you need to define the field as "percentage". But this requires that the field should be of a certain value. For example, a 0.1 value is displayed at 10%. So, all the percentage fields need to be lower than 1 and as a consequence of type float in Elasticsearch. So, we need to define all floats in the Elasticsearch template.

### CODE REVIEW

**Comment by monicasarbu** [comment]
Fixing bug #29

**Comment by monicasarbu** [comment]
Fixing bug #31

**Comment by ruflin** [comment]
@monicasarbu Seems like some of the tests are failing because they expect 0.95 instead of 95%

File: docs/fields.asciidoc [hunk 1]
```diff
@@ -0,0 +1,438 @@
+
+////
+This file is generated! See etc/fields.yml and scripts/generate_field_docs.py
+////
+
+[[exported-fields]]
+== Exported fields
+
+This document describes the fields that are exported by the
+Packetbeat shipper for each transaction. They are grouped in the
```
Comment (ruflin) [inline docs/fields.asciidoc]: Topbeat

File: scripts/generate_template.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+
+"""
+This script generates the ES template file (packetbeat.template.json) from
```
Comment (ruflin) [inline scripts/generate_template.py]: topbeat

File: scripts/generate_template.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+the etc/fields.yml file.
+
+Example usage:
+
+   python generate_template.py etc/fields.yml etc/packetbeat.template.json
```
Comment (ruflin) [inline scripts/generate_template.py]: topbeat

File: scripts/generate_field_docs.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    output.write("""
+////
+This file is generated! See etc/fields.yml and scripts/generate_field_docs.py
+////
+
+[[exported-fields]]
+== Exported fields
+
+This document describes the fields that are exported by the
+Packetbeat shipper for each transaction. They are grouped in the
```
Comment (ruflin) [inline scripts/generate_field_docs.py]: Topbeat

File: etc/topbeat.template.json [hunk 5]
```diff
@@ -31,4 +99,4 @@
     "index.refresh_interval": "5s"
   },
   "template": "topbeat-*"
```
Comment (ruflin) [inline etc/topbeat.template.json]: Not sure but should we have here also the new "template" format?

Comment (monicasarbu) [inline etc/topbeat.template.json]: not needed here in the template, but we need to change the index in kibana (unfortunately manually for now).

**Comment by monicasarbu** [comment]
Keep the PR open in order to remove "." from the topbeat exported fields. As it's not supported by ES 2.0

File: docs/fields.asciidoc [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+type: int
+
+Virtual memory the process has in total.
+
+
+==== proc.mem.rss
+
+type: int
+
+Resident Set Size. Memory occupied by the process in main memory(RAM).
```
Comment (andrewkroh) [inline docs/fields.asciidoc]: `s/memory(RAM)/memory (RAM)/` ?
Occurs in a few places.

File: etc/fields.yml [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+              path: proc.mem.size
+              type: int
+              description: >
+                Virtual memory the process has in total.
+
+            - name: rss
+              path: proc.mem.rss
+              type: int
+              description: >
+                Resident Set Size. Memory occupied by the process in main memory(RAM).
```
Comment (andrewkroh) [inline etc/fields.yml]: `s/memory(RAM)/memory (RAM)/` ?
Occurs in a few places.

**Comment by andrewkroh** [comment]
LGTM

File: etc/fields.yml [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+          type: int
+          description: >
+            Actual available memory. Available only on Unix.
+
+    - name: swap
+      type: group
+      description: This group contains statistics related to the swap memory usage on the system.
+      fields:
+        - name: total
+          path: swap.total
```
Comment (ruflin) [inline etc/fields.yml]: @monicasarbu These dots don't matter?

Comment (monicasarbu) [inline etc/fields.yml]: Yes, they do matter. "proc.name": "bash" is the same thing as "proc": {"name": "bash"}, but only the latter works in ES 2.0.

**Pull request merged**